src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -326,26 +324,36 @@ private static SymbolInfo GetSymbolInfo(SemanticModel semanticModel, SyntaxNode
                         // In some cases a single clause binds to more than one method. In those cases 
                         // the tokens in the clause determine which of the two SymbolInfos are returned.
                         // See also the proposal at https://github.com/dotnet/roslyn/issues/23394
-                        return token.IsKind(SyntaxKind.InKeyword) ? queryInfo.CastInfo : queryInfo.OperationInfo;
+                        return token.IsKind(SyntaxKind.InKeyword) ? queryInfo.CastInfo.GetBestOrAllSymbols() : queryInfo.OperationInfo.GetBestOrAllSymbols();
                     }
 
                     if (hasCastInfo)
-                    {
-                        return queryInfo.CastInfo;
-                    }
+                        return queryInfo.CastInfo.GetBestOrAllSymbols();
 
-                    return queryInfo.OperationInfo;
+                    return queryInfo.OperationInfo.GetBestOrAllSymbols();
                 case IdentifierNameSyntax { Parent: PrimaryConstructorBaseTypeSyntax baseType }:
-                    return semanticModel.GetSymbolInfo(baseType, cancellationToken);
+                    return semanticModel.GetSymbolInfo(baseType, cancellationToken).GetBestOrAllSymbols();
             }
 
             //Only in the orderby clause a comma can bind to a symbol.
             if (token.IsKind(SyntaxKind.CommaToken))
+                return ImmutableArray<ISymbol>.Empty;
+
+            // If we're on 'var' then asking for the symbol-info will get us the symbol *without* nullability
+            // information. Check for that, and try to return the type with nullability info if it has it.
+            if (node is IdentifierNameSyntax { IsVar: true })
             {
-                return default;
+                var symbol = semanticModel.GetSymbolInfo(node, cancellationToken).GetAnySymbol();
+                var type = semanticModel.GetTypeInfo(node, cancellationToken).Type;
+                if (type != null &&
+                    type.Equals(symbol, SymbolEqualityComparer.Default) &&
+                    !type.Equals(symbol, SymbolEqualityComparer.IncludeNullability))
```

Comment:
❔ What happens if we make this unconditional?

Reply:
I would prefer to keep it conditional so that we're not changing from symbolinfo to itypesymbol for the general case (where they are the same), but only when the itypesymbol provides additional information not present in the original.